dspace-server-webapp/src/main/java/org/dspace/app/rest/BundleUploadBitstreamController.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.dspace.app.rest.utils.RegexUtils.REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.SQLException;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.converter.BitstreamConverter;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.hateoas.BitstreamResource;
+import org.dspace.app.rest.repository.BundleRestRepository;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Bundle;
+import org.dspace.content.service.BundleService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ControllerUtils;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.hateoas.ResourceSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+/**
+ * Controller to upload bitstreams to a certain bundle, indicated by a uuid in the request
+ * Usage: POST /api/core/bundles/{uuid}/bitstreams (with file and properties of file in request)
+ * Example:
+ * <pre>
+ * {@code
+ * curl https://<dspace.server.url>/api/core/bundles/d3599177-0408-403b-9f8d-d300edd79edb/bitstreams
+ *  -XPOST -H 'Content-Type: multipart/form-data' \
+ *  -H 'Authorization: Bearer eyJhbGciOiJI...' \
+ *  -F "file=@Downloads/test.html" \
+ *  -F 'properties={ "name": "test.html", "metadata": { "dc.description": [ { "value": "example file", "language": null,
+ *          "authority": null, "confidence": -1, "place": 0 } ]}, "bundleName": "ORIGINAL" };type=application/json'
+ * }
+ * </pre>
+ */
+@RestController
+@RequestMapping("/api/core/bundles/" + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID)
```

Comment:
Same as above, this RequestMapping should be *more specific*.  It should be: `@RequestMapping("/api/core/bundles/" + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID + "/bitstreams")` as it is *specifically* for `/api/core/bundles/{uuid}/bitstreams` requests.

As noted in the above comment, this also ensures we are not creating *two controllers* with the same RequestMapping.  Doing so is NOT recommended as it can result in confusing (and difficult to track down bugs) in the future.  Again, see comment above for more info too.